lxd/endpoints/network_util.go

```diff
@@ -0,0 +1,47 @@
+package endpoints
+
+import (
+	"net"
+	"os"
+	"regexp"
+)
+
+type networkServerErrorLogWriter struct {
+	proxies []net.IP
+}
+
+// Regex for the log we want to ignore.
+var unwantedLogRegex = regexp.MustCompile(`http: TLS handshake error from ([^\[:]*|\[(.*?)]):.*write: connection reset by peer\s$`)
```

Comment:
I'm not familiar with the regex implementation in Go but the `\[` seems unbalanced to me. Also, in Stéphane's log, there was no white space before the end of the line.

With `egrep` here is what I came up with:

````
echo 'Sep 17 04:58:33 abydos lxd.daemon[21884]: 2021/09/17 04:58:33 http: TLS handshake error from [2602:fd23:8:1002:216:3eff:fe13:6bb1]:55672: write tcp [2602:fd23:8:101::100]:8443->[2602:fd23:8:1002:216:3eff:fe13:6bb1]:55672: write: connection reset by peer' | egrep 'http: TLS handshake error from (\[[^]]+\]|[0-9.]+):[0-9]+: .+ write: connection reset by peer$'
````

From there, `\1` should contain the IPv4 or the IPv6 with enclosing `[]`, without the port. If the `[]` should be dropped, there might be a regex way to do it, just let me know.

Reply:
The first option in the outer/first capture group greedily matches anything that is not `[` or `:`, so for an IPv4 address it gets everything before the colon which delimits it from the port. The second option in the outer/first capture group matches the `[` starting an IPv6 address, then lazily matches anything until a closing `]` is reached (this doesn't need to be escaped because there is no preceding unescaped opening bracket - though I agree it looks a little odd so I'm happy to escape it anyway if you prefer).

Regarding white space, the problem is that the regex isn't operating directly against what is logged by the `http.Server`. It is mutated by the `log.Logger` first, prepending the time and date and adding a newline, and then passed into the `io.Writer` we've created. (I've just updated the tests to reflect this.)

I'm not sure if I've explained this well at all so I've written a little go playground to illustrate it: https://go.dev/play/p/bE2enDF1uIr.

Reply:
Thanks for providing the go playground link, that was instructive. I like the regex you came up with, but would however like to see the closing `]` escaped for extra clarity, if you don't mind.

Reply:
Oh, and what would you think of replacing the `*` by `+` since those inside the capturing groups are required to be present?

Reply:
Good idea, it's now the same as in your go playground :+1: